### ISSUE TITLE
Cosmo DB connection

### ISSUE THREAD

**Issue by pechaut73** [dgkanatsios/AzureServicesForUnity#12]
Hi,

I tried to do as you said in the command, put 
TableStorageClient.Instance.EndpointStorageType = EndpointStorageType.CosmosDBTableAPI;

and

 TableStorageClient.Instance.AuthenticationToken = "AccountEndpoint=https://cosmosdb-prague.documents.azure.com:443/;AccountKey=UU6OXXXXXXXXXXXXXXXXXXXXX

I launch : no error, but if I try to interrogate the DB I get an error, saying that it can not parse a webserver adress",

any idea on what I'm doing wrong ?

CHeers,
PE

**Comment by dgkanatsios** [comment]
Are you using CosmosDB with Table API? If yes, you should use the AuthenticationToken credential by setting it with your AccountKey.

https://social.msdn.microsoft.com/Forums/SECURITY/en-US/e993cf4b-41e9-4fa3-bd1a-b871f737f95b/where-is-accountkey-for-cosmosdb-in-azure-portal?forum=azurecosmosdb

**Comment by pechaut78** [comment]
Actually, the key has to be edited.
The account key should be removed and only the account name should be set, so no HTTPS decoration etc..

But even then:

if my URI is marked as: https://MYNAME.documents.azure.com:443/
I set: TableStorageClient.Instance.SetAccountName("MYNAME");

it creates a connection to: MYNAME.table.cosmosdb.azure.com/
and when hitting CreateIfNoExist, it says : can not resolve host..

is table.cosmosdb.azure.com/ correct ? The URI on the azure key page is :  https://MYNAME.documents.azure.com:443/

**Comment by dgkanatsios** [comment]
> is table.cosmosdb.azure.com/ correct ? The URI on the azure key page is : https://MYNAME.documents.azure.com:443/

this is the URI when Cosmos DB is used with "SQL" API. We want Cosmos DB to be used with the "Table" API. https://docs.microsoft.com/en-us/azure/cosmos-db/table-introduction

**Comment by pechaut78** [comment]
Ok, I see, but MYNAME.table.cosmosdb.azure.com/Pages() generates a host can not be resolved. If I try it in a webbrowser it says the IP adress can not be found..

**Comment by dgkanatsios** [comment]
I don't think you're supposed to call it via an unauthenticated HTTP call via a web browser. Have you tried using the CosmosDB SDK?

**Comment by pechaut78** [comment]
YEs exactly, using the app it says "can not resolve destination host" when clicking on CreateIfNoExist

In the log a connection to URL= https:/MYNAME.table.cosmosdb.azure.com/Tables()
then 0
then Cannot resolve destination host

Normally I should see an invalid request with no authorization in the browser..

**Comment by dgkanatsios** [comment]
the SDK in the Unity app works only with Cosmos DB configure with the Table API, not with the SQL API, so the `myname.table.cosmosdb.azure.com` url is not supposed to work.

**Comment by pechaut78** [comment]
I see, but I get:

EndpointStorageType.TableStorage :
URL= https://iot-prague.table.core.windows.net/Tables()
Cannot resolve destination host

EndpointStorageType.CosmosDBTableAPI :
https://iot-prague.table.cosmosdb.azure.com/Tables()
Cannot resolve destination host

Same result

**Comment by dgkanatsios** [comment]
Totally missed that, sorry. Did you manage to make it work?

**Comment by GnutekG** [comment]
Hey! I have exactly the same issue. "Cannot resolve destination host". The URL that this tool generates for me looks like this:
 https://ACCOUNT_NAME.table.cosmosdb.azure.com/Routes()?$select=locationName,startName,endName

Now it occured to me: there's a place for "tableName" parameter in TableStorageClient.QueryTable, what is that name if we have database A with containers B & C in it? I guess container == table, but what would happen if we had more than one database under the same account name?
